gt/main.c

```diff
@@ -633,6 +732,16 @@ cleanup_gt_instance(struct gt_instance *instance)
 {
 	lua_close(instance->lua_state);
 	instance->lua_state = NULL;
+
+	if (instance->neigh.cache_tbl) {
+		rte_free(instance->neigh.cache_tbl);
+		instance->neigh.cache_tbl = NULL;
+	}
+
+	if (instance->neigh.hash_table) {
+		rte_hash_free(instance->neigh.hash_table);
+		instance->neigh.hash_table = NULL;
+	}
```

Comment:
Since the Lua state is initialized before the cache table and hash table, I think the closing of the Lua state should come after the freeing of these tables.